storage/src/mmr/journaled.rs

```diff
@@ -384,15 +386,20 @@ impl<E: RStorage + Clock + Metrics, H: CHasher> Mmr<E, H> {
         }
         let mut mem_mmr = MemMmr::init(MemConfig {
             nodes: vec![],
-            pruned_to_pos: journal_size,
+            pruned_to_pos: Position::new(journal_size),
             pinned_nodes: mem_pinned_nodes,
             pool: cfg.config.thread_pool,
         });
 
         // Add the additional pinned nodes required for the pruning boundary, if applicable.
         if cfg.lower_bound_pos < journal_size {
-            Self::add_extra_pinned_nodes(&mut mem_mmr, &metadata, &journal, cfg.lower_bound_pos)
-                .await?;
+            Self::add_extra_pinned_nodes(
+                &mut mem_mmr,
+                &metadata,
+                &journal,
+                cfg.lower_bound_pos.as_u64(),
```

Comment:
Can we make add_extra_pinned_nodes to take a Position?

Reply:
Done